cargo-pgx/README.md

```diff
@@ -686,3 +686,77 @@ OPTIONS:
     -V, --version
             Print version information
 ```
+
+## EXPERIMENTAL: Versioned shared-object support
+
+`pgx` experimentally supports the option to produce a versioned shared library. This allows multiple versions of the
+extension to be installed side-by-side, and can enable the deprecation (and removal) of functions between extension
+versions. There are some caveats which must be observed when using this functionality. For this reason it is currently
+experimental.
+
+### Activation
+
+Versioned shared-object support is enabled by removing the `module_pathname` configuration value in the extension's
+`.control` file.
+
+### Concepts
+
+Postgres has the implicit requirement that C extensions maintain ABI compatibility between versions. The idea behind
+this feature is to allow interoperability between two versions of an extension when the new version is not ABI
+compatible with the old version.
+
+The mechanism of operation is to version the name of the shared library file, and to hard-code function definitions to
+point to the versioned shared library file. Without versioned shared-object support, the SQL definition of a C function
+would look as follows:
+
+```SQL
+CREATE OR REPLACE FUNCTION "hello_extension"() RETURNS text /* &str */
+STRICT
+LANGUAGE c /* Rust */
+AS 'MODULE_PATHNAME', 'hello_extension_wrapper';
+```
+
+`MODULE_PATHNAME` is replaced by Postgres with the configured value in the `.control` file. For pgx-based extensions,
+this is  usually set to `$libdir/<extension-name>`.
+
+When using versioned shared-object support, the same SQL would look as follows:
+
+```SQL
+CREATE OR REPLACE FUNCTION "hello_extension"() RETURNS text /* &str */
+STRICT
+LANGUAGE c /* Rust */
+AS '$libdir/extension-0.0.0', 'hello_extension_wrapper';
+```
```

Comment:
Does versioned shared-object support also make the output `.so` have a different name?

Reply:
> Does versioned shared-object support also make the output .so have a different name?

Yup

Reply:
I've mentioned it here in the readme.